### PR TITLE
azure blob storage persistentstorage adapter, unit tests

### PR DESCRIPTION
**Pull request by yatli**

### CODE REVIEW

File: src/Modules/Trinity.DynamicCluster/Trinity.Azure.Storage.Test/DownloadTest.cs [hunk 1]
```diff
@@ -37,13 +50,21 @@ public void Init()
...
         [TestMethod]
-        public async Task GetLatestVersion()
+        public async Task Download()
         {
-            var v = await m_storage.GetLatestVersion();
-            Assert.AreEqual(m_version, v);
+            GetLatestVersion().Wait();
+            var v = await m_storage.Download(m_version, 0, 110);
+            var src = await v.DownloadAsync();
+            var src_seq = src.GetEnumerator();
```
Comment (yatli) [inline src/Modules/Trinity.DynamicCluster/Trinity.Azure.Storage.Test/DownloadTest.cs]: ps: GetEnumerator() can be implicitly invoked if you do `foreach(var buf in src_seq)`

Comment (yatli) [reply src/Modules/Trinity.DynamicCluster/Trinity.Azure.Storage.Test/DownloadTest.cs]: and thus you could say `Assert.IsTrue(Enumerable.SequentialEqual(src_seq.Single().First(), TestItem`

Comment (thautwarm) [reply src/Modules/Trinity.DynamicCluster/Trinity.Azure.Storage.Test/DownloadTest.cs]: Need help!  
This one passed:
```C#
var src_seq = src.GetEnumerator();
var buffer = src_seq.Current.Buffer;
Assert.IsTrue(buffer.Take(data.Length).SequenceEqual(data.Take(data.Length)));
```
However, this one cannnot work well.
```C#
var buffer = src.First().Buffer;
Assert.IsTrue(buffer.Take(data.Length).SequenceEqual(data.Take(data.Length)));
```
The second one occurs ``System.AccessViolationException at [PersistedCellEnumerator.cs, line 32](https://github.com/Microsoft/GraphEngine/blob/availability/src/Modules/Trinity.DynamicCluster/Trinity.DynamicCluster/Persistency/PersistedCellEnumerator.cs)

**Review by yatli — COMMENTED** [review]
LGTM. @thautwarm could you please leave a comment how to repro the known access violation issue? it could be a faulty pointer on my side.

**Pull request merged**